### DISCUSSION TITLE
Deploy Backend First and then Frontend

### DISCUSSION THREAD

**Discussion by maksymlumen** [fluxcd/flux2#3971] [Q&A]
Hi everyone. 
I am using monorepo approach with backend and frontend inside it. Changes in folders trigger images builds. Then ImageRepo + Image Policy + Image Automation -  update two separate helm releases values with updated image tag. 

Everything works great except sometimes frontend being updated earlier and users may try to access to features which are not yet rolled out on a backend.
Does anyone know what approach should I take to ensure that one automation happens after another. 
Thank you!

**Comment by makkes** [comment]
Hi @maksymlumen, this is a great question and a common issue. In your specific situation what you can do to mitigate the frontend being upgraded prior to the backend:

1. Let Flux's ImageUpdateAutomation push to a branch from which PRs are created (e.g. with GitHub Actions or similar automation) and wait for all commits to arrive (frontend and backend), then merge.
1. Wrap the HelmReleases in a Kustomization each and add a [dependency](https://fluxcd.io/flux/components/kustomize/kustomization/#dependencies) to the frontend's Kustomization on the backend's Kustomization.

**Reply by maksymlumen** [reply]
Thanks @makkes. I will try out dependency approach first.

**Reply by makkes** [reply]
Both need to work together, it's not either 1 or 2, it's 1 and 2.

**Reply by maksymlumen** [reply]
oh, I see, thanks
